packages/eas-cli/src/credentials/android/utils/keystoreNew.ts

```diff
@@ -0,0 +1,91 @@
+import { getX509Asn1ByFriendlyName, parsePKCS12 } from '@expo/pkcs12';
+import jks from 'jks-js';
+import { asn1 } from 'node-forge';
+
+import { AndroidKeystoreType } from '../../../graphql/generated';
+import Log from '../../../log';
+import { Keystore, KeystoreWithType } from '../credentials';
+
+export function getKeystoreWithType(keystore: Keystore): KeystoreWithType {
+  const type = getKeystoreType(keystore);
+  return { ...keystore, type };
+}
+
+export function getKeystoreType(keystore: Keystore): AndroidKeystoreType {
+  if (isPKCSKeystore(keystore)) {
+    return AndroidKeystoreType.Pkcs12;
+  } else if (isJKSKeystore(keystore)) {
+    return AndroidKeystoreType.Jks;
+  }
+  return AndroidKeystoreType.Unknown;
```

Comment:
:(

Reply:
there's unfortunately a long tail of obscure keystore formats (bks, jceks, etc), and not a good way to detect what file type a keystore is. it is hard to determine if its the obscure keystore or just a corrupt one :/